kubernetes/kubernetes.go

```diff
@@ -186,6 +187,41 @@ func CollectKubeMetrics(config KubeAgentConfig) {
 
 }
 
+func performConnectionChecks(ka *KubeAgentConfig) {
+	log.Info("Performing connectivity checks. Checking that the agent can retrieve S3 URL")
+
+	cldyMetricClient, err := client.NewHTTPMetricClient(client.Configuration{
+		Token:         ka.APIKey,
+		Verbose:       false,
+		ProxyURL:      ka.OutboundProxyURL,
+		ProxyAuth:     ka.OutboundProxyAuth,
+		ProxyInsecure: ka.OutboundProxyInsecure,
+	})
+	if err != nil {
+		log.Fatalf("error creating Cloudability Metric client: %v ", err)
+	}
+
+	metricSampleURL := apiEndpoint + "/metricsample"
+
+	file, err := os.Create("/tmp/temp.txt")
```

Comment:
https://pkg.go.dev/os#CreateTemp

Also is good for this.

Either way we should clean it up as a best practice.

```
defer os.Remove(file.Name())
```

Reply:
Good call, added defer line